### PR TITLE
Raspberry Pi 3: init

### PR DESCRIPTION
**Pull request by pinage404**
###### Description of changes

Upstream my config for Raspberry PI 3

It's just some config that make it works for me

It's heavily inspired by the file that create sd-image for Aarch64

I don't understand everything of it

###### Things done

- [x] Tested the changes in your own NixOS Configuration
- [x] [Tested the changes end-to-end by using your fork of `nixos-hardware` and importing it via `<nixos-hardware>` or Flake input](https://gitlab.com/pinage404/dotfiles/-/commit/2ffa280243d2f667df762e3337e7236c7666fde5)

### CODE REVIEW

File: raspberry-pi/3/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  # The serial ports listed here are:
+  # - ttyS0: for Tegra (Jetson TX1)
+  # - ttyAMA0: for QEMU's -machine virt
+  # https://github.com/NixOS/nixpkgs/blob/b72bde7c4a1f9c9bf1a161f0c267186ce3c6483c/nixos/modules/installer/sd-card/sd-image-aarch64.nix#L19
+  boot.kernelParams = [
+    "console=ttyS0,115200n8"
+    "console=ttyAMA0,115200n8"
+    "console=tty0"
+  ];
```
Comment (Mic92) [inline raspberry-pi/3/default.nix:33]:
```suggestion
  # The serial ports listed here are:
  # - ttyS0: serial
  # - tty0: hdmi
  boot.kernelParams = [
    "console=ttyS0,115200n8"
    "console=tty0"
  ];
```

File: raspberry-pi/3/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  environment.systemPackages = with pkgs; [
+    libraspberrypi
```
Comment (Mic92) [inline raspberry-pi/3/default.nix:36]: Can you add a comment, what this package provides to the user?

Comment (pinage404) [reply raspberry-pi/3/default.nix:36]: I don't remember why i added it 😅

File: raspberry-pi/3/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  ];
```
Comment (Mic92) [inline raspberry-pi/3/default.nix:37]: Alright, than we can drop it from the default configuration. Until we know why we need it.

```suggestion
```

Comment (Mic92) [reply raspberry-pi/3/default.nix:37]: https://github.com/NixOS/nixos-hardware/pull/1016/files#r1656670399

File: raspberry-pi/3/default.nix (hunk 1 above)
Comment (Mic92) [reply raspberry-pi/3/default.nix:33]: @pinage404 does your pi also shows tty0 even if there is no hdmi connected?
I am trying to figure out what the best default value is.

Comment (pinage404) [reply raspberry-pi/3/default.nix:33]: With no HDMI plug-in

```sh
ll /dev/tty0
crw--w---- 1 root tty 4, 0  6 déc.   2023 /dev/tty0
```

With the current config

```nix
  boot.kernelParams = [
    "console=ttyS0,115200n8"
    "console=ttyAMA0,115200n8"
    "console=tty0"
  ];
```

Comment (Mic92) [reply raspberry-pi/3/default.nix:33]: I don't think we ever need /dev/ttyAMA0 on raspberry pi. This device is qemu specific.
However printing kernel logs on hdmi is probably what most people expect. The configuration that I propose allows access via serial, but prints kernel logs to hdmi.

**Review by Mic92 — APPROVED** [review]
Looks good from my side. Should we merge it like this?

File: raspberry-pi/3/default.nix (hunk 1 above)
Comment (pinage404) [reply raspberry-pi/3/default.nix]: i don't understand `boot.kernelParams`, it is black magic for me

**Comment by pinage404** [comment]
Seems good for me !

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**